src/hotspot/share/opto/superword.cpp

```diff
@@ -3397,6 +3389,32 @@ LoadNode::ControlDependency SuperWord::control_dependency(Node_List* p) {
   return dep;
 }
 
+// Find the memop pack with the maximum vector width, unless they were already
+// determined by SuperWord::filter_packs_for_alignment().
+void SuperWord::determine_mem_ref_and_aw_for_main_loop_alignment() {
+  if (_mem_ref_for_main_loop_alignment != nullptr) {
+    assert(vectors_should_be_aligned(), "mem_ref only set if filtered for alignment");
+    return;
+  }
+
+  MemNode const* mem_ref = nullptr;
+  int max_aw = 0;
+  for (int i = 0; i < _packset.length(); i++) {
+    Node_List* pack = _packset.at(i);
+    MemNode* first = pack->at(0)->isa_Mem();
+    if (first == nullptr) { continue; }
+
+    int vw = first->memory_size() * pack->size();
```

Comment:
I assume `first` is verified already and `first->memory_size()` is reasonable (size of primitive type).
So how `bottom[]` fits into this?

Reply:
Yes, it is. All of this code is run in `SuperWord::output`, and at this point we are committed to vectorization - everything is verified.

Reply:
That is what I tried to say in the PR description:
> It makes more sense to pick a mem_ref directly in SuperWord::adjust_pre_loop_limit_to_align_main_loop_vectors, where we already know what packs are going to be vectorized.

Reply:
Yes, `first->memory_size()` knows the size in bytes of the load/store. It is used many places in SuperWord.